src/proxy/topic/TopicProxy.ts

```diff
@@ -0,0 +1,53 @@
+import {BaseProxy} from "../BaseProxy";
+import {ITopic} from "../ITopic";
+import {MessageListener} from "../MessageListener";
+import {UUID} from "../../core";
+import {TopicAddMessageListenerCodec} from "../../codec/TopicAddMessageListenerCodec";
+import {TopicRemoveMessageListenerCodec} from "../../codec/TopicRemoveMessageListenerCodec";
+import {TopicPublishCodec} from "../../codec/TopicPublishCodec";
+import { ListenerMessageCodec} from "../../listener/ListenerMessageCodec";
+import {ClientMessage, ClientMessageHandler} from "../../protocol/ClientMessage";
+import {assertNotNull} from "../../util/Util";
+
+export class TopicProxy<E> extends BaseProxy implements ITopic<E> {
+
+    publish(message: E): Promise<void> {
+        assertNotNull(message);
+        const data = this.toData(message);
+        const request = TopicPublishCodec.encodeRequest(this.name, data);
+        const partitionId = request.getPartitionId();
+        return this.encodeInvokeOnPartition<void>(TopicAddMessageListenerCodec,  partitionId, TopicAddMessageListenerCodec.decodeResponse);
+    }
+
+    addMessageListener(listener: MessageListener<E>): string {
+        assertNotNull(listener);
+        const message = TopicAddMessageListenerCodec.encodeRequest(this.name, false);
+        const codec = this.createItemListener(this.name, false)
+        TopicAddMessageListenerCodec.handle(message, () => {});
+        //I am unable to fix this part. Will be trying tomorrow
+        return this.listenerService.registerListener(codec,  (message => {
+            TopicAddMessageListenerCodec.handle(message, ()=> {})
+        }));
+    }
+
+    removeMessageListener(listenerId: string): boolean {
+        if (this.listenerService.deregisterListener(listenerId)){
```

Comment:
you can simplify this by removing if statement.

Reply:
`TS2322: Type 'Promise<boolean>' is not assignable to type 'boolean'.` I did in this way. Any other way to solve?

Reply:
Ok this is some real problematic thing. ITopic.removeMessageListener is added synchronously because in reliable topic it can be synchronous but in ITopic it cannot be. Changing ITopic is a breaking change. I don't know what we should do. Let us talk in the standup tomorrow.